packages/shared/src/components/modals/award/GiveAwardModal.tsx

```diff
@@ -123,10 +143,53 @@ const IntroScreen = () => {
         </Typography>
         <div className="mt-4 grid grid-cols-3 gap-2 tablet:grid-cols-4">
           {awards?.edges?.map(({ node: item }) => (
-            <AwardItem key={item.id} item={item} />
+            <AwardItem
+              key={item.id}
+              item={item}
+              onClick={({ product: clickedProduct }) => {
+                if (clickedProduct.value > user.balance.amount) {
+                  setActiveStep({ screen: 'INTRO', product: clickedProduct });
+                  setShowBuyCores(true);
+
+                  return;
+                }
+
+                setShowBuyCores(false);
+                setActiveStep({ screen: 'COMMENT', product: clickedProduct });
+              }}
+            />
           ))}
         </div>
       </Modal.Body>
+      {showBuyCores && !!product && (
+        <Modal.Footer className="!h-auto flex-col" justify={Justify.Center}>
+          <Button
+            className="w-full"
+            variant={ButtonVariant.Primary}
+            onClick={onBuyCores}
+          >
+            Buy Cores <CoinIcon />{' '}
+            {product.value === 0 ? 'Free' : product.value}
+          </Button>
+          <Typography
+            type={TypographyType.Footnote}
+            color={TypographyColor.Quaternary}
+            className="text-center"
+          >
+            Awards may include a revenue share with the recipient and are
+            subject to our{' '}
+            <a
+              href={termsOfService}
+              target="_blank"
+              rel="noopener"
```

Comment:
I think you can use defaultAnchorRel here (or something like that)

Reply:
done